src/gluonts/model/statsforecast.py

```diff
@@ -0,0 +1,232 @@
+
+from dataclasses import dataclass, field
+from typing import List, Optional, Type
+
+import numpy as np
+
+from statsforecast.models import (
+    ADIDA,
+    AutoARIMA,
+    AutoCES,
+    CrostonClassic,
+    CrostonOptimized,
+    CrostonSBA,
+    ETS,
+    IMAPA,
+    TSB,
+)
+
+from gluonts.core.component import validated
+from gluonts.dataset import DataEntry
+from gluonts.dataset.util import forecast_start
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.model.forecast import QuantileForecast
+
+
+@dataclass
+class ModelConfig:
+    quantile_levels: Optional[List[float]] = None
+    forecast_keys: List[str] = field(init=False)
+    statsforecast_keys: List[str] = field(init=False)
+    intervals: Optional[List[int]] = field(init=False)
+
+    def __post_init__(self):
+        self.forecast_keys = ["mean"]
+        self.statsforecast_keys = ["mean"]
+        if self.quantile_levels is None:
+            self.intervals = None
+            return
+
+        intervals = set()
+
+        for quantile_level in self.quantile_levels:
+            interval = round(
+                200 * (max(quantile_level, 1 - quantile_level) - 0.5)
+            )
+            intervals.add(interval)
+            side = "hi" if quantile_level > 0.5 else "lo"
+            self.forecast_keys.append(str(quantile_level))
+            self.statsforecast_keys.append(f"{side}-{interval}")
+
+        self.intervals = sorted(intervals)
+
+
+class StatsForecastPredictor(RepresentablePredictor):
+    """
+    A predictor type that wraps models from the `statsforecast`_ package.
+
+    This class is used via subclassing and setting the ``ModelType`` class
+    attribute to specify the ``statsforecast`` model type to use.
+
+    .. _statsforecast: https://github.com/Nixtla/statsforecast
+
+    Parameters
+    ----------
+    prediction_length
+        Prediction length for the model to use.
+    quantile_levels
+        Optional list of quantile levels that we want predictions for.
+        By default this is ``None``, giving only the mean predition.
+    **kwargs
+        Keyword arguments to be passed to the model type for construction.
+        The specific arguments accepted or required depend on the
+        ``ModelType``; please refer to the documentation of ``statsforecast``
+        for details.
+    """
+
+    ModelType: Type
+
+    @validated()
+    def __init__(
+        self,
+        prediction_length: int,
+        quantile_levels: Optional[List[float]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(prediction_length=prediction_length)
+        self.model = self.ModelType(**kwargs)
+        self.config = ModelConfig(quantile_levels=quantile_levels)
+
+    def predict_item(self, entry: DataEntry) -> QuantileForecast:
+        # TODO use also exogenous features
+        kwargs = {}
+        if self.config.intervals is not None:
+            kwargs["level"] = self.config.intervals
```

Comment:
Is there a neutral value for `level`?

Reply:
This suggest we could just use `None`.

https://github.com/Nixtla/statsforecast/blob/fecbfc31c6cba3d615147565196f7be1459d5a90/statsforecast/models.py#L278

Reply:
the thing is, only arima accepts this `level` argument, so for all other models one should just not set the quantiles and `level` not be passed

Reply:
But if only arima takes it, why do we expose it for all methods?

Reply:
Mainly https://github.com/Nixtla/statsforecast/issues/248#issuecomment-1256539527